### ISSUE TITLE
First2Finish track

### ISSUE THREAD

**Issue by lstkz** [topcoderinc/cab#58]
There are many challenges with prize >$300, and I think the current format doesn't work well for many members.

1. Timezone issue.
Example:
The challenge starts at 10 PM, and when I wake up on the next day, I can see 5 submissions or the challenge is complete.
I couldn't participate in any F2F from cloudhub or drone series projects.
I believe many members from similar timezone have the same problem.

2. Reviewer
Sometimes you must wait 6h - 8h for the reviewer. This problem is related to 1. issue.
If you live in the same timezone with the reviewer, you can get 5 iterations quickly.

3. Risk
If there is already 1 submission, some members don't join the contest because there is a risk that our submission won't be reviewed. Some members try to submit something fast, and after 1-2h they resubmit a better submission.

Suggested solution.
We can convert F2F to fast 24h Code challenges with only 1 reviewer and 1 prize.
I think the current F2F is only good for <$50 contests. We don't have such contests because copilots/PMs prefer bug bashes.

**Comment by ThomasKranitsas** [comment]
@lsentkiewicz all your above points are valid but I think that in most cases this is based on client's budget/decision because running a CODE challenge is way more expensive than running a F2F.

**Comment by iversonLv** [comment]
Totally agree Point 2.
I think the title should be "Review" not "Reviewer" :)
Additional I think about Point 2, if the reviewer 'know' the submitter, then he/she can last the review timeline, then let the submitter ( his/her) friend to implement another submission for another iteration.

E.g.
A(Reviewer) and B are team
A review C sub
A found C some issues but A does not submit the scorecard immediately but wait for B sub. Now C is waiting for the result who does not know whether his/her sub win or lose.
Then After B finish new sub, A rejects C, B submits.

If A, B, C are independent, A review C quickly and reject, maybe C and B are at same level to fix/sub new sub..

So reviewer could control the timeline between submission if he/she want?

What is you guys thoughts, maybe communities discuss such case?

**Comment by iversonLv** [comment]
CODE challenge need $600(maybe now change) for copilot?
F2F is $40?
Budget gap is large, client/ game plan are not permit I believe.

**Comment by iversonLv** [comment]
And another thought is, default F2F review payment is low, but reviewer could get more/much payment via rejecting submissions, maybe found a few issue one iteration by one?

That is the case as well.
E.g.
A review B v1, he/she found some issue base on B, but he/she does not list all, he/she rejects the submission with some, leave some?
B sub v2 iteration, A lists v1 remaining issues + v2 some, reject again,
..
...
..
Maybe could get lots of submissions($$$)

**Comment by iversonLv** [comment]
3. That is why F2F :0
Can we bring something like, if A submit in 1-2hr, if he is rejected, then he could resubmit a X hr gap(said punishment hr)?

Then we can save reviewer time, another is block too much low quality submissions?
there is a risk now, if I submit a low quality submission, my next submit time need duration X hr, I must take care of my every submission.

**Comment by birdofpreyru** [comment]
> I think the current F2F is only good for <$50 contests. We don't have such contests because copilots/PMs prefer bug bashes.

I agree. F2F with larger price tag usually involve some larger workload, and too high risk, that the outcome won't be much correlated with the job done by you. I personally mess with them only when they relate to a well-familiar project, and I've caught the moment they were just started, so the chance to win is relatively higher.

> We can convert F2F to fast 24h Code challenges with only 1 reviewer and 1 prize.

I believe it won't help much. The challenge with 24h timeline won't be too challenging. If few experienced members go for it, all will deliver equally good solutions, different in tiny issues, and you know, I don't like the situation when decision on the winner comes to the small differences.

I believe, the way to go for that intermediate challenges (say > $100, but < $500 jobs) is the direct tasks (or how they are called in the discussion in Slack), when a member says `I'll do it in this time` and goes to do it without competition. But for such challenges, I'd like to see a fair system to distribute such tasks so that everybody who is willing and qualified to participate can get the same money out of them in transparent way (i.e. that not all such challenges go to a few members just because the co-pilot knows them from past projects and already very comfy working with them).

**Comment by iversonLv** [comment]
In fact I like Bug Bounty system v.s. F2F, members assign tickets on their own at one time.
Finally, get $$$ bonus who get most ticket amount.
Members could select easy task with low prize with risk that might not get final $$$.

**Comment by lstkz** [comment]
@ThomasKranitsas 
The total cost would be the same.
Only 1st prize
Only 1 reviewer
And the copilot fee is the same


The main idea is that all submissions are guaranteed to be reviewed.

**Comment by sumitdaga** [comment]
The current F2F can be tweaked a little may be, such that all submissions in the first 24 hrs will be treated equally, as in all will be reviewed and the best submission(obviously which fulfils all requirements) will be judged winner and after 24 hrs, it can go back to being "first to finish the tasks" kind .

Also 24 hrs need not be strict, it can depend on the how big/small the task is ....the co-pilot can just mention it in the challenge spec ...for ex all submissions in the first 8 hrs will be treated equally. By default it can be 24 hrs.

**Comment by veshu** [comment]
Instead of tweaking or restrcting current format. It is better to run f2f at specific time so that timezone issue will minimized. E.g. allow to post f2f only four times a day.  The thing i like f2f is when i can start from beginning a rush to finish early makes you more concentrate and focus please dont change the core f2f format. For timezone issue of reviewer cant we have f2f reviwer as well☺

**Comment by hokienick** [comment]
CAB MEETING MARCH

Will be pulling in Mess

**Comment by sumitdaga** [comment]
I have a general suggestion not specific to the F2F track.
How about mention the timezone of the co-pilot for a challenge either in the forums or on the challenge page .
But possibly the co-pilot for whatever reason works at odd times, like late night, etc ...so, if there could be an indicator to suggest if the co-pilot is available right now and if not then tentatively when will he/she be available next.
It may help in all the tracks. As a submitter you would have at least a rough idea when your questions will be answered

**Comment by hokienick** [comment]
process issue, suggestion is to spec review but that may slow down the process.  Need more thought.
